camel/models/nexa_model.py

```diff
@@ -0,0 +1,173 @@
+                for the model. If not provided, `OpenAITokenCounter(ModelType.
+                GPT_4O_MINI)` will be used.
+        """
+        self.model_type = model_type
+        self.model_config_dict = model_config_dict
+        # TODO: Cannot start server now
+        # Should be started by user, and url should be provided
+        if not url and not os.environ.get("NEXA_BASE_URL"):
+            self._start_server()
+        self._url = (
+            url
+            or os.environ.get("NEXA_BASE_URL")
+            or "http://localhost:8000/v1"
+        )
+        if not self._url.endswith("/v1"):
+            self._url += "/v1"
+        self._api_key = api_key or "DO-NOT-USE-API-KEY"
+        self._token_counter = token_counter
+        self.check_model_config()
+        self._client = OpenAI(
+            timeout=60,
+            max_retries=3,
+            base_url=self._url,
+            api_key=self._api_key,
+        )
+
+    def _start_server(self) -> None:
+        """Starts the Nexa server in a subprocess."""
+        try:
+            subprocess.Popen(
+                [
+                    "nexa",
+                    "server",
+                    "llama3.2",
+                    "--port",
+                    "8000",
+                ],  # TODO: model type
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+            )
+            print(
+                f"Nexa server started on http://localhost:8000 "
+                f"for {self.model_type} model."
+            )
+        except Exception as e:
+            print(f"Failed to start Nexa server: {e}.")
+
+    def check_model_config(self):
+        """Check whether the model configuration contains any
+        unexpected arguments to Nexa API.
+
+        Raises:
+            ValueError: If the model configuration dictionary contains any
+                unexpected arguments to Nexa API.
+        """
+        for param in self.model_config_dict:
+            if param not in NEXA_API_PARAMS:
+                raise ValueError(
+                    f"Unexpected argument `{param}` is "
+                    "input into Nexa model backend."
+                )
+
+    def run(
+        self,
+        messages: List[OpenAIMessage],
+    ) -> Union[ChatCompletion, Stream[ChatCompletionChunk]]:
+        """Runs inference of Nexa chat completion.
+
+        Args:
+            messages (List[OpenAIMessage]): Message list with the chat history
+                in OpenAI API format.
+
+        Returns:
+            Union[ChatCompletion, Stream[ChatCompletionChunk]]:
+                `ChatCompletion` in the non-stream mode, or
+                `Stream[ChatCompletionChunk]` in the stream mode.
+        """
+        try:
+            response = self._client.chat.completions.create(
+                messages=messages,
+                # model=self.model_type,
+                model="model-type",
```

Comment:
I think here shouldn't be hard coded

Reply:
Same question as above. This would be solved if I simply use `str` type.